### PR TITLE
chart: Update chart for the enhanced backing image feature

### PR DESCRIPTION
**Pull request by shuo-wu**
Longhorn #2295

### CODE REVIEW

**Comment by shuo-wu** [comment]
Will update this PR when a tag like `v20210405` is created for `longhornio/backing-image-manager`

**Comment by innobead** [comment]
@shuo-wu please rebase the code and remove the draft for ready-for-review. Thanks.

**Comment by shuo-wu** [comment]
Updated. I have tested it. It's ready for review.

File: chart/values.yaml [hunk 1]
```diff
@@ -22,6 +22,9 @@ image:
     shareManager:
       repository: longhornio/longhorn-share-manager
       tag: v1_20210406
+    backingImageManager:
+      repository: longhornio/backing-image-manager
+      tag: master
```
Comment (innobead) [inline chart/values.yaml:27]: need to have a tag for the image instead of master

File: chart/questions.yml [hunk 2]
```diff
@@ -70,6 +70,18 @@ questions:
...
     label: Longhorn Share Manager Image Tag
     group: "Longhorn Images Settings"
+  - variable: image.longhorn.backingImageManager.repository
+    default: longhornio/backing-image-manager
+    description: "Specify Longhorn Backing Image Manager Image Repository"
+    type: string
+    label: Longhorn Backing Image Manager Image Repository
+    group: "Longhorn Images Settings"
+  - variable: image.longhorn.backingImageManager.tag
+    default: master
```
Comment (innobead) [inline chart/questions.yml:80]: same here

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just need to update the image tag.

**Review by innobead — APPROVED** [review]

File: chart/templates/crds.yaml [hunk 3]
```diff
@@ -417,3 +417,54 @@ spec:
...
+      openAPIV3Schema:
+        type: object
+        properties:
+          spec:
+            x-kubernetes-preserve-unknown-fields: true
+          status:
+            x-kubernetes-preserve-unknown-fields: true
+    subresources:
+      status: {}
+    additionalPrinterColumns:
```
Comment (PhanLe1010) [inline chart/templates/crds.yaml:451]: Do we want to include the  `.status.currentState` as other CRD?

**Review by PhanLe1010 — COMMENTED** [review]
LGTM. 
Left a comment

File: chart/templates/crds.yaml (hunk 3 above)
Comment (shuo-wu) [reply chart/templates/crds.yaml:451]: It's nice to have. Let me add it with a new PR.

File: chart/templates/crds.yaml [hunk 4]
```diff
@@ -417,3 +417,54 @@ spec:
...
+    - name: Image
+      type: string
+      description: The image the manager pod will use
+      jsonPath: .spec.image
```
Comment (PhanLe1010) [inline chart/templates/crds.yaml:455]: A quick quesition, any reason that we prefer `.spec.image` instead of `.status.currentImage` as other CRDs?

Comment (shuo-wu) [reply chart/templates/crds.yaml:455]: when I use a newer version of the backing image manager, we will directly create new CRs for the new version. In other words, we won't modify `spec.image` then we don't need `status.currentImage`. Same for InstanceManager and ShareManager.

Comment (PhanLe1010) [reply chart/templates/crds.yaml:455]: I see. Thanks

**Pull request merged**